src/Game.cpp

```diff
@@ -27,24 +28,62 @@ void Game::setIsStrikeOrSpare(unsigned short int position, const std::string& va
     }
 }
 
-int Game::checkValue(const std::string & value)
+int Game::checkValue(const std::string& value)
 {
-    int valueInt;
-    auto valueTemp=value.substr(0,2);
-    if ((value[0] == 'X') || (value[1] == '/'))
+    int firstBall, secondBall;
+    std::tie(firstBall, secondBall) = convertValueToInt(value);
+    return firstBall + secondBall;
+}
+
+std::pair <int, int> Game::convertValueToInt(const std::string& value)
+{
+    int valInt1 = 0;
+    int valInt2 = 0;
+
+    auto valueTemp = value.substr(0, 2);
+    std::replace(valueTemp.begin(), valueTemp.end(), '-', '0');
+
+    if (value[0] == 'X') valInt1 = 10;
+    else if (value[1] == '/') 
```

Comment:
It should be check `else if (value.size() == 2 && value[1] == '/')`, because if will be no full frame can be problem